spec/user_interface_spec.rb

```diff
@@ -221,6 +220,67 @@
     end
   end
 
+  describe '#search_term' do
+    it 'prints a prompt asking user for a search term' do
+      input = StringIO.new("john\n")
+      ui = described_class.new(input, output, validator)
+
+      ui.search_term
+
+      expect(output.string).to include(described_class::SEARCH_MESSAGE)
+    end
+
+    it 'returns the search term typed by user' do
+      input = StringIO.new("john\n")
+      ui = described_class.new(input, output, validator)
+
+      expect(ui.search_term).to eq('john')
+    end
+
+    it 'prints error message if no input is given' do
+      input = StringIO.new("\njohn\n")
+      ui = described_class.new(input, output, validator)
+
+      ui.search_term
+
+      expect(output.string).to include(described_class::ERROR_MESSAGE)
+    end
+  end
+
+  describe '#search_again?' do
+    it 'prints a prompt to user asking to search for another contact' do
+      input = StringIO.new(yes_reply)
+      ui = described_class.new(input, output, validator)
+
+      ui.search_again?
+
+      expect(output.string).to include(described_class::ANOTHER_SEARCH_PROMPT)
+    end
+
+    it 'returns true if user wants to search another contact' do
+      input = StringIO.new(yes_reply)
+      ui = described_class.new(input, output, validator)
+
+      expect(ui.search_again?).to eq(true)
+    end
+
+    it 'returns false if user doesnt want to search another contact' do
+      input = StringIO.new('n')
```

Comment:
doesn't this need a `\n` at the end to represent reality? 

until the user presses enter, the method `gets` can not read the value, so in this test `search_again?` is false just by accident.